File: src/net/merayen/elastic/backend/data/storage/StorageView.java
package net.merayen.elastic.backend.data.storage;

import java.util.List;

public interface StorageView extends AutoCloseable {
	public StorageFile readFile(String path);
	public StorageFile writeFile(String path);
	public boolean exists(String path);
	public List<String> list(String path);
	public List<String> listAll(String path);
	public void close();
}

File: src/net/merayen/elastic/backend/data/storage/StorageView.kt
package net.merayen.elastic.backend.data.storage

interface StorageView : AutoCloseable {
    fun readFile(path: String): StorageFile
    fun writeFile(path: String): StorageFile
    fun exists(path: String): Boolean
    fun list(path: String): List<String>
    fun listAll(path: String): List<String>
    override fun close()
}
